File: IE307-server/src/models/Account.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;
const ObjectId=Schema.ObjectId;

const Product = new Schema({
    productId: { type: ObjectId },
    name: { type: String, required: true, max: 255 },
    description: { type: String, max: 800 },
    image: { type: String, max: 255 },
    price: { type: String, default: 0 },
    type: { type: String, max: 25 },
    material: { type: String, max: 25, default: 'Silver' },
    createdAt: { type: Date, default: Date.now },
    updatedAt: { type: Date, default: Date.now },
});

const Account = new Schema({
    userId:{
        type: ObjectId,
    },
    username: {
        type: String,
        required: true,
        unique: true,
        max: 50,
    },
    password: {
        type: String,
        required: true,
        max: 30,
    },
    role:{
        type: String,
        default: "USER",
    },
    email: {
        type: String,
        default: "none@gmail.com",
    },
    phone: {
        type: String,
    },
    address: {
        type: String,
    }, 
    favorite: [{
        item: Product,
    }],

},{
    timestamps: true,
});

module.exports = mongoose.model("Account", Account);

File: IE307-client/IE307/Models/Account.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System;
using System.Collections.Generic;
using System.Text;
using Xamarin.Forms;

namespace IE307.Models
{
   
    public class Favorite
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string itemId { get; set; }
        [BsonElement("item")]
        public Product item { get; set; }
    }
    public class Account
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string userId { get; set; }
        public string username { get; set; }
        public string password { get; set; }    
        public string role { get; set; }    
        public string email { get; set; }   
        public string phone { get; set; }
        public string address { get; set; }
        [BsonElement("favorite")]
        public List<Favorite> favorite { get; set; }   
        [BsonElement("createdAt")]
        [BsonRepresentation(BsonType.DateTime)]
        public DateTime createdAt { get; set; }
        [BsonElement("updatedAt")]
        [BsonRepresentation(BsonType.DateTime)]
        public DateTime updatedAt { get; set; }
        [BsonElement("__v")]
        public int __v { get; set; }
    }
}
